manufacturing-client/src/main.rs

```diff
@@ -296,10 +301,79 @@ async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType>
     }
     match mfg_string_type {
         MfgStringType::SerialNumber => {
-            let serial = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+            log::debug!(
+                "msg_string_type value:{:#?} refers SerialNumber format ",
+                mfg_string_type
+            );
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
                 .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial))
+            Ok(CborSimpleType::Text(serial_dmi))
+        }
+        MfgStringType::SerialNumberExt => {
+            log::debug!(
+                "msf_string_type value:{:#?} refers SerialNumberExt format ",
+                mfg_string_type
+            );
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+                .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
+                .map(|s| s.trim_end().to_owned())
+                .context("Error determining system serial number")?;
```

Comment:
It might make sense to trim `serial_dmi` over here

Reply:
yes ,it does not need trim later in the code . Its done here